### ISSUE TITLE
[BUG]: WCAG20_Input_ExplicitLabel does not support role="checkbox" with label as child

### ISSUE THREAD

**Issue by tombrunet** [IBMa/equal-access#756]
### Project

a11y checker engine

### Description

See https://act-rules.github.io/testcases/e086e5/e860f8badc29cd5db9b60050f73c395a0a8de519.html
This testcase should pass since ARIA 1.2 lists that checkbox can get label from 'contents':
https://www.w3.org/TR/wai-aria-1.2/#checkbox

I think this is only true when using an explicit `role="checkbox"`.

Note: it appears the same is true for role="radio".

Related to ACT rule [e086e5](https://act-rules.github.io/rules/e086e5)

### Steps to reproduce

Run ACT testcase: https://act-rules.github.io/testcases/e086e5/e860f8badc29cd5db9b60050f73c395a0a8de519.html
See violation: "Form control with "checkbox" role has no associated label"

This should not be triggering.

**Comment by tombrunet** [comment]
See also https://github.com/IBMa/equal-access/issues/761

**Comment by shunguoy** [comment]
The ACT rule [e086e5] (https://act-rules.github.io/rules/e086e5) is not a label, but rather accessible names as a whole. We should use this to check all the accessible names.

**Comment by shunguoy** [comment]
this issue will be fixed in the rule WCAG20_Input_ExplicitLabel, and a different issue is open: https://github.com/IBMa/equal-access/issues/844 to handle the more generic cases.
